src/dataflow/components/dataflow-program.tsx

```diff
@@ -641,6 +651,28 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     return nodePos;
   }
 
+  private moveNodeToFront = (node: any, newNode: boolean) => {
+    const totalNodes = this.programEditor.nodes.length;
+    const selectedNodeView = this.programEditor.view.nodes.get(node);
+    let selectedNodeZ = 0;
+    if (selectedNodeView) {
+      selectedNodeZ = selectedNodeView.el.style.zIndex ? parseInt(selectedNodeView.el.style.zIndex, 10) : selectedNodeZ;
+    }
+    this.programEditor.nodes.forEach((n: Node) => {
+      const nodeView = this.programEditor.view.nodes.get(n);
+      if (nodeView) {
+        if (node.id === n.id) {
+          nodeView.el.style.zIndex = totalNodes.toString();
+        } else if (nodeView.el.style.zIndex) {
+          const nodeZ = parseInt(nodeView.el.style.zIndex, 10);
+          if (nodeZ > selectedNodeZ && !newNode) {
```

Comment:
Out of curiosity is the `newNode` argument required here? Wouldn't a new node have a z-index of 0 in which case we shouldn't get here?

Reply:
I'm trying to put the new node on the top actually.  This let's me know now to subtract 1 from the z-indices of the existing nodes, but to simply plop the new one on top using the total number of nodes to get its z-index.